### ISSUE TITLE
How to implement in app browser?

### ISSUE THREAD

**Issue by 1hbb** [OAuthSwift/OAuthSwift#646]
### Description:
This code works perfect but i want to open authorize page internal browser. How can i achieve this ? 
```swift

let oauthswift = OAuth2Swift(
            consumerKey: "**********",
            consumerSecret: "***********",
            authorizeUrl: "https://www.reddit.com/api/v1/authorize.compact",
            accessTokenUrl: "https://www.reddit.com/api/v1/access_token",
            responseType: "code",
            contentType: "application/json"
        )
        self.oauthswift = oauthswift
        oauthswift.accessTokenBasicAuthentification = true

        let state = generateState(withLength: 20)
        _ = oauthswift.authorize(
            withCallbackURL: URL(string: "SwiftReddit://oauth-callback/reddit")!, scope: "read,identity,edit,flair,history,modconfig,modflair,modlog,modposts,modwiki,mysubreddits,privatemessages,report,save,submit,subscribe,vote,wikiedit,wikiread", state: state, parameters: ["duration": "permanent"]
        ) { result in
            switch result {
            case .success(let (credential, _, _)):
                print("Reddit Toke: ", credential.oauthToken)
            case .failure(let error):
                print(error.description)
            }
        }

```

### OAuth Version:
- [ ] Version 1
- [x] Version 2

### OS (Please fill the version) :
- [x] iOS :
- [ ] OSX :
- [ ] TVOS :
- [ ] WatchOS  :

### Installation method:
- [ ] Carthage
- [ ] CocoaPods
- [x] Swift Package Manager
- [ ] Manually

### Library version:
- [ ] head
- [x] v2.1.0
- [ ] v2.0.0
- [ ] v1.4.1
- [ ] other: (Please fill in the version you are using.)

### Xcode version:
- [x] 11.4 (Swift 5.2)
- [ ] 11.x (Swift 5.1)
- [ ] 10.x (Swift 5.0)
- [ ] other: (Please fill in the version you are using.)

- [ ] objective c

**Comment by phimage** [comment]
internal browser?

custom handler doc could be found here 
https://github.com/OAuthSwift/OAuthSwift#handle-authorize-url

in App Store internal webview for auth is not welcomed, prefer use SafariURLHandler

**Comment by 1hbb** [comment]
> internal browser?

I'm sorry that I misrepresented it. I meant aswebauthenticationsession. Can u give little working example. I searched all the demo app source code but could not find a clear example.

**Comment by 1hbb** [comment]
I got it like below. Thanks for this great package 👍🏻

```swift
func getURLHandler() -> OAuthSwiftURLHandlerType {
        if #available(iOS 9.0, *) {
            let handler = SafariURLHandler(viewController: self, oauthSwift: self.oauthswift!)
            handler.presentCompletion = {
                print("Safari presented")
            }
            handler.dismissCompletion = {
                print("Safari dismissed")
            }
            return handler
        }
        return OAuthSwiftOpenURLExternally.sharedInstance
    }

oauthswift.authorizeURLHandler = getURLHandler()

```
